pkg/config/loader.go

```diff
@@ -105,36 +124,60 @@ func ParseWinPDTargets(log logr.Logger, vpcCniConfigMap *v1.ConfigMap) (warmIPTa
 		warmPrefixTargetStr, foundWarmPrefix = vpcCniConfigMap.Data[WinWarmPrefixTarget]
 	}
 
-	// If no configuration is found, return 0
-	if !foundWarmIP && !foundMinIP && !foundWarmPrefix {
-		return warmIPTarget, minIPTarget, warmPrefixTarget
-	}
-
+	// If warm IP target config value is not found, or there is an error parsing it, the value will be set to zero
 	if foundWarmIP {
-		warmIPTargetInt, err := strconv.Atoi(warmIPTargetStr)
+		warmIPTarget, err = strconv.Atoi(warmIPTargetStr)
 		if err != nil {
-			log.Error(err, "failed to parse warm ip target", "warm ip target", warmIPTargetStr)
-		} else {
-			warmIPTarget = warmIPTargetInt
+			log.Info("Could not parse warm ip target, defaulting to zero", "warm ip target", warmIPTargetStr)
+		} else if !isPDEnabled && warmIPTarget == 0 {
+			// Handle secondary IP mode scenario where WarmIPTarget is explicitly configured to zero
+			// In such a case there must always be 1 warm IP to ensure that the warmpool is never empty
+			log.V(1).Info("Explicitly setting WarmIPTarget zero value not supported in secondary IP mode, will override with 1")
+			warmIPTarget = 1
 		}
+	} else {
+		log.V(1).Info("could not find warm ip target in ConfigMap, defaulting to zero")
```

Comment:
nit: same here. It could be hard for us to track back historic config in the configmap audit logs if we need help customers to identify the cause.